great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -131,9 +179,225 @@ def _get_domains(
+            parameters=None,
+        )
+        exclude_semantic_types = _parse_semantic_domain_type_argument(
+            semantic_types=exclude_semantic_types
+        )
+
+        if not (self.include_semantic_types or self.exclude_semantic_types):
+            return build_simple_domains_from_column_names(
+                column_names=table_column_names,
+                domain_type=self.domain_type,
+            )
+
+        column_types_dict_list: List[Dict[str, Any]] = validator.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.column_types",
+                metric_domain_kwargs={
+                    "batch_id": batch_ids[-1],  # active_batch_id
+                },
+                metric_value_kwargs={
+                    "include_nested": True,
+                },
+                metric_dependencies=None,
+            )
+        )
+
+        column_name: str
+
+        table_column_name_to_inferred_semantic_domain_type_mapping: Dict[
+            str, SemanticDomainTypes
+        ] = {
+            column_name: self.infer_semantic_domain_type_from_table_column_type(
+                column_types_dict_list=column_types_dict_list,
+                column_name=column_name,
+            ).semantic_domain_type
+            for column_name in table_column_names
+        }
+
+        candidate_column_names: List[str] = copy.deepcopy(table_column_names)
+
+        if include_semantic_types:
+            candidate_column_names = list(
+                filter(
+                    lambda candidate_column_name: table_column_name_to_inferred_semantic_domain_type_mapping[
+                        candidate_column_name
+                    ]
+                    in include_semantic_types,
+                    candidate_column_names,
+                )
+            )
+
+        if exclude_semantic_types:
+            candidate_column_names = list(
+                filter(
+                    lambda candidate_column_name: table_column_name_to_inferred_semantic_domain_type_mapping[
+                        candidate_column_name
+                    ]
+                    not in exclude_semantic_types,
+                    candidate_column_names,
+                )
+            )
+
+        domains: List[Domain] = [
+            Domain(
+                domain_type=self.domain_type,
+                domain_kwargs={
+                    "column": column_name,
+                },
+                details={
+                    "inferred_semantic_domain_type": table_column_name_to_inferred_semantic_domain_type_mapping[
+                        column_name
+                    ],
+                },
+            )
+            for column_name in candidate_column_names
+        ]
+
+        return domains
+
+    # This method (default implementation) can be overwritten (with different implementation mechanisms) by subclasses.
+    @staticmethod
+    def infer_semantic_domain_type_from_table_column_type(
```

Comment:
In a future refactor could some of this functionality be moved to a separate `ColumnSemanticTypeFilter` or similar?

Reply:
@anthonyburdi Yes.  Since this did not make the release, I might just follow up on all of your feedback carrying action items this morning...  Thanks!